source/common/router/upstream_request.cc

```diff
@@ -526,6 +541,7 @@ void UpstreamRequest::onPoolReady(
         [this]() -> void { onStreamMaxDurationReached(); });
     max_stream_duration_timer_->enableTimer(*max_stream_duration);
   }
+  copyDynamicMetaDataFromL4Downstream(info);
```

Comment:
Will add the downstream connection id as the key when copy the metadata to upstream so that upstream can include all metadata from different downstream.